### PR TITLE
Support transfer ObjectRef from JAVA application to PYTHON deployment

### PR DESCRIPTION
**Pull request by yucai**
## Why are these changes needed?

In the current implementation, it's not possible to transfer an ObjectRef from a Java application to a Python deployment. For more details, please refer to this issue: https://github.com/ray-project/ray/issues/45676

**Root Cause**
In the original approach, all arguments were bundled together and passed as a list parameter, like so: {1, 2, "parameter3"}. If one of these parameters was an ObjectRef, it would lead to serialization issues, preventing cross-language transfer: {1, 2, "parameter3", objectRef4}.

**Proposed Solution**
In the new approach, each argument is passed independently. The ObjectRef is implemented as PassByReference, which avoids the previous serialization issues and enables the transfer of ObjectRef: {1, 2, "parameter3", objectRef4}. 

This change enhances the interoperability between Java applications and Python deployments in Ray.

### CODE REVIEW

**Comment by yucai** [comment]
This issue is currently a significant roadblock preventing us from moving our **1st Ray-based use case into a production** environment. Could anyone assist by reviewing this? Your help would be greatly appreciated.

File: java/api/src/main/java/io/ray/api/PyActorCall.java [hunk 1]
```diff
@@ -40,4 +40,10 @@ default <R> PyActorTaskCaller<R> task(PyActorMethod<R> pyActorMethod, Object obj
     return new PyActorTaskCaller<>((PyActorHandle)this, pyActorMethod, args);
   }
 
+  default <R> PyActorTaskCaller<R> flatten_task(PyActorMethod<R> pyActorMethod, Object obj0, Object[] obj1) {
```
Comment (SongGuyang) [inline java/api/src/main/java/io/ray/api/PyActorCall.java:43]: Seems a new API ".flatten_task" is added here. Can we avoid api changes and fix this issue by changing the logic of the existing ".task" API ?

Comment (yucai) [reply java/api/src/main/java/io/ray/api/PyActorCall.java]: @SongGuyang I have removed the new function. Kindly share your thoughts and suggestions with me.

Comment (yucai) [reply java/api/src/main/java/io/ray/api/PyActorCall.java]: @SongGuyang have passed all tests.
https://buildkite.com/ray-project/premerge/builds/26727

**Comment by yucai** [comment]
eBay is currently undertaking a POC with Ray that is of critical importance to our company's AI platform strategy. We would greatly appreciate it if you could review the PR at your earliest convenience. 

@SongGuyang @kfstorm @jjyao @stephanie-wang

**Comment by liuyang-my** [comment]
LGTM

**Review by SongGuyang — APPROVED** [review]
LGTM, thanks for this contribution!

**Pull request merged**